client/src/context/useKanbanContext.tsx

```diff
@@ -22,32 +32,40 @@ export const KanbanProvider: FunctionComponent = ({ children }): JSX.Element =>
   const [focusedCard, setFocusedCard] = useState<ICard | null>(null);
   const { updateSnackBarMessage } = useSnackBar();
   const { loggedInUser } = useAuth();
+  const history = useHistory();
 
   useEffect(() => {
-    if (loggedInUser) getFirstBoard();
-
+    if (loggedInUser) getAllBoards();
     return;
-  }, [loggedInUser]);
+  }, [history, loggedInUser]);
 
-  const getFirstBoard = async (): Promise<IBoard> => {
-    const request = await getUserBoards();
-    const board = request.boards[0];
-    setActiveBoard(board);
-    setUserBoards(request.boards);
-    setColumns(board.columns);
+  const sendToFirstBoard = useCallback(async () => {
+    const { boards } = await getUserBoards();
+
+    if (boards) {
+      history.push(`/dashboard/board/${boards[0]._id}`);
+      return;
+    }
+    history.push(`/newboard`);
+  }, []);
 
-    return board;
+  const getAllBoards = async (): Promise<void> => {
+    const { boards } = await getUserBoards();
+    setUserBoards(boards);
   };
 
-  const fetchBoard = async (id: string): Promise<IBoard> => {
-    const board = await getBoard(id);
+  const fetchBoard = async (id: string): Promise<void> => {
+    const { board } = await getBoard(id);
```

Comment:
On my environment this shows an error relating to the promise response of getBoard sending over a `IBoard` however Board does not have `board` property.

Reply:
I have changed the interface for this call, it should not give you this error. I'll double check.

Reply:
![image](https://user-images.githubusercontent.com/39752144/120906247-fd861300-c614-11eb-8dda-7fd752a229b3.png) Sometimes I need to restart react for it to adapt to new types, maybe that happened to you.

Reply:
Yeah, looks like my TypeScript intellisense was being slow. No worries!